src/fidesops/service/masking/strategy/masking_strategy_factory.py

```diff
@@ -10,70 +10,57 @@
-from fidesops.service.masking.strategy.masking_strategy_nullify import (
-    NullMaskingStrategy,
-)
-from fidesops.service.masking.strategy.masking_strategy_random_string_rewrite import (
-    RandomStringRewriteMaskingStrategy,
-)
-from fidesops.service.masking.strategy.masking_strategy_string_rewrite import (
-    StringRewriteMaskingStrategy,
-)
 
+logger = logging.getLogger(__name__)
 
-class SupportedMaskingStrategies(Enum):
-    """
-    The supported methods by which Fidesops can pseudonymize data.
-    """
 
-    string_rewrite = StringRewriteMaskingStrategy
-    random_string_rewrite = RandomStringRewriteMaskingStrategy
-    null_rewrite = NullMaskingStrategy
-    hash = HashMaskingStrategy
-    aes_encrypt = AesEncryptionMaskingStrategy
-    hmac = HmacMaskingStrategy
+class MaskingStrategyFactory:
+    registry: Dict[str, MaskingStrategy] = {}
+    valid_strategies = ""
 
     @classmethod
-    def __contains__(cls, item: str) -> bool:
-        try:
-            cls[item]
-        except KeyError:
-            return False
+    def register(cls, name: str) -> Callable[[MaskingStrategy], MaskingStrategy]:
+        def wrapper(strategy_class: MaskingStrategy) -> MaskingStrategy:
+            if name in cls.registry:
+                logger.warning(
+                    f"Strategy with name {name} already exists in MaskingStrategy registry. It referred to class {cls.registry[name]}, but will now refer to {strategy_class}"
+                )
+            logger.info(
+                f"Registering MaskingStrategy class {strategy_class} under name {name}"
+            )
+            cls.registry[name] = strategy_class
+            cls.valid_strategies = ", ".join(cls.registry.keys())
+            return strategy_class
 
-        return True
-
-
-def get_strategy(
-    strategy_name: str,
-    configuration: Dict[
-        str,
-        Union[str, FormatPreservationConfig],
-    ],
-) -> MaskingStrategy:
-    """
-    Returns the strategy given the name and configuration.
-    Raises NoSuchStrategyException if the strategy does not exist
-    """
-    if not SupportedMaskingStrategies.__contains__(strategy_name):
-        valid_strategies = ", ".join([s.name for s in SupportedMaskingStrategies])
-        raise NoSuchStrategyException(
-            f"Strategy '{strategy_name}' does not exist. Valid strategies are [{valid_strategies}]"
-        )
-    strategy = SupportedMaskingStrategies[strategy_name].value
-    try:
-        strategy_config: MaskingConfiguration = strategy.get_configuration_model()(
-            **configuration
-        )
-        return strategy(configuration=strategy_config)
-    except ValidationError as e:
-        raise FidesopsValidationError(message=str(e))
+        return wrapper
 
+    @classmethod
+    def get_strategy(
+        cls,
+        strategy_name: str,
+        configuration: Dict[
+            str,
+            Union[str, FormatPreservationConfig],
+        ],
```

Comment:
This formatting is 🤮 
```suggestion
        configuration: Dict[str, Union[str, FormatPreservationConfig]],
```

Reply:
hmmm...pretty sure this was the auto-formatter, maybe i've got some settings that are off?

Reply:
If you remove the trailing command on L45, it'll reformat this way.

Reply:
oh right, duh